### ISSUE TITLE
Stop recording LSL

### ISSUE THREAD

**Issue by roberto9110** [sccn/labstreaminglayer#344]
I am trying to crate a Matlab GUI in which, using a toggle button, I can stop the recording from an EEG amplifier.
How can I do that?
Thanks

**Comment by cboulay** [comment]
How are you recording the data to begin with? Are you using LabRecorder?

**Comment by roberto9110** [comment]
Yes indeed
________________________________

**Comment by roberto9110** [comment]
Ok. It is something will be possible to develop in the future
Thanks
________________________________

**Comment by dmedine** [comment]
Controlling an amplifier itself from Matlab (or wherever) is not something that LSL does. This is the job of your hardware device's SDK. 

Back in the day, LabRecorder was scriptable because it was implemented in Python. It got re-done in C++ sometime ago. For a while I was thinking of going back to the python implementation (the bugs in liblsl-Python got hammered out) but since then a fair amount of improvement has been made. 

One idea would be to implement a kind of REST-style protocol for messaging via LSL marker streams. This would be a way to expose any LSL app's functionality to pretty much anything. This requires a lot of careful consideration and design.

**Comment by roberto9110** [comment]
But, using the 'while' in matlab, is it possible to manage this?
Thanks


________________________________
Da: Chadwick Boulay <notifications@github.com>
Inviato: martedì 11 settembre 2018 07:24
A: sccn/labstreaminglayer
Cc: roberto9110; Author
Oggetto: Re: [sccn/labstreaminglayer] Stop recording LSL (#344)


I was thinking about that too. My thought-experiment solutions were a bit awkward in that anytime an app wanted to send a command to LabRecorder it would have to REcreate the control stream to clobber the old one and take over control. Would these control apps then have to destroy their outlets? Would LabRecorder be able to handle the disappearance and reappearance of the "same" stream from different apps?

**Comment by dmedine** [comment]
@tstenner You are right of course about the consumer/producer inversion. I was thinking of this as a kind of stop-gap just for LabRecorder. It could have a listening thread that could spawn inlets to listen for the pre-defined commands (like 'stop'). Realistically no one would ever try to connect a bunch of outlets to control LabRecorder. Lua bindings is certainly more desirable.
